lib/src/coap_message.dart

```diff
@@ -122,25 +125,18 @@ class CoapMessage {
   }
 
   /// Returns the first option of the specified type, or null
-  CoapOption? getFirstOption(int optionType) {
-    final options = getOptions(optionType) as List<CoapOption>?;
-    if (options == null) {
-      return null;
-    }
-    for (final option in options) {
-      if (option.type == optionType) {
-        return option;
-      }
-    }
-    return null;
+  CoapOption? getFirstOption(OptionType optionType) {
+    return getOptions(optionType)
+        ?.toList()
+        .firstWhere((element) => element.type == optionType);
```

Comment:
Maybe we have to use `firstWhereOrNull` here, `firstWhere` will throw an exception if it doesn't find a match:

> If no element satisfies test, the result of invoking the orElse function is returned. If orElse is omitted, it defaults to throwing a [StateError](https://api.dart.dev/stable/2.0.0/dart-core/StateError-class.html).

https://api.dart.dev/stable/2.0.0/dart-core/Iterable/firstWhere.html

Reply:
Good catch, fixed in 7648235c38be2a02a2bda7c35c10a21d05afe761 :)